pubtools/pulplib/_impl/fake/rpmlib.py

```diff
@@ -25,4 +25,75 @@ def broken(*_args, **_kwargs):
     get_header_fields = broken
     get_keys_from_header = broken
 
-__all__ = ["get_rpm_header", "get_header_fields", "get_keys_from_header"]
+__all__ = [
+    "get_rpm_header",
+    "get_header_fields",
+    "get_keys_from_header",
+    "get_rpm_requires",
+    "get_rpm_provides",
+]
+
+
+def get_rpm_requires(header):
+    header_flags_map = {
+        "version": "REQUIREVERSION",
+        "name": "REQUIRENAME",
+        "flags": "REQUIREFLAGS",
+    }
+    return _get_rpm_deps(header, header_flags_map)
+
+
+def get_rpm_provides(header):
+    header_flags_map = {
+        "version": "PROVIDEVERSION",
+        "name": "PROVIDENAME",
+        "flags": "PROVIDEFLAGS",
+    }
+    return _get_rpm_deps(header, header_flags_map)
+
+
+def _get_rpm_deps(header, header_flags_map):
+    header_flags = list(header_flags_map.values())
+    raw_deps = get_header_fields(header, header_flags)
+
+    names = raw_deps[header_flags_map["name"]]
+    versions = raw_deps[header_flags_map["version"]]
+    flags = raw_deps[header_flags_map["flags"]]
+
+    deps = []
+
+    for name, vr, flag in zip(names, versions, flags):
+        version, release = vr.split("-")
+
+        deps_item = {
+            "name": name,
+            "version": version,
+            "release": release,
+            "epoch": None,  # unsure how to get this value from rpm headers
```

Comment:
I think this ought to be fixed.  I looked for an RPM with non-zero epoch dependency and I think this info is also in the REQUIREVERSION field?

```
$ rpm -qp FreeWnn-devel-1.10pl020-6.el4.x86_64.rpm --requires
warning: FreeWnn-devel-1.10pl020-6.el4.x86_64.rpm: Header V3 DSA/SHA1 Signature, key ID db42a60e: NOKEY
FreeWnn-libs = 1:1.10pl020-6.el4
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1

$ rpm -qp FreeWnn-devel-1.10pl020-6.el4.x86_64.rpm --queryformat '%{REQUIREVERSION}'
warning: FreeWnn-devel-1.10pl020-6.el4.x86_64.rpm: Header V3 DSA/SHA1 Signature, key ID db42a60e: NOKEY
1:1.10pl020-6.el4
```

So I think the "vr" variable right now is actually "evr", with the "e" component frequently omitted.